src/main/java/org/spongepowered/common/mixin/api/text/MixinText.java

```diff
@@ -24,116 +24,68 @@
  */
 package org.spongepowered.common.mixin.api.text;
 
-import com.google.common.collect.ImmutableList;
-import net.minecraft.util.text.ITextComponent;
-import net.minecraft.util.text.Style;
-import net.minecraft.util.text.TextComponentBase;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.spongepowered.api.text.LiteralText;
 import org.spongepowered.api.text.Text;
-import org.spongepowered.api.text.action.ClickAction;
-import org.spongepowered.api.text.action.HoverAction;
-import org.spongepowered.api.text.action.ShiftClickAction;
-import org.spongepowered.api.text.format.TextColors;
-import org.spongepowered.api.text.format.TextFormat;
-import org.spongepowered.asm.mixin.Final;
 import org.spongepowered.asm.mixin.Mixin;
-import org.spongepowered.asm.mixin.Shadow;
-import org.spongepowered.common.interfaces.text.IMixinTextComponent;
-import org.spongepowered.common.interfaces.text.IMixinText;
-import org.spongepowered.common.text.action.SpongeClickAction;
-import org.spongepowered.common.text.action.SpongeHoverAction;
-import org.spongepowered.common.text.format.SpongeTextColor;
-
-import java.util.Optional;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.common.text.impl.LiteralTextImpl;
 
 @Mixin(value = Text.class, remap = false)
-public abstract class MixinText implements IMixinText {
-
-    @Shadow @Final protected TextFormat format;
-    @Shadow @Final protected ImmutableList<Text> children;
-    @Shadow @Final protected Optional<ClickAction<?>> clickAction;
-    @Shadow @Final protected Optional<HoverAction<?>> hoverAction;
-    @Shadow @Final protected Optional<ShiftClickAction<?>> shiftClickAction;
-
-    private ITextComponent component;
-    private String json;
-
-    protected TextComponentBase createComponent() {
-        throw new UnsupportedOperationException();
+public interface MixinText {
```

Comment:
One goal of moving Text implementation to SpongeCommon is to avoid having mixins for our own API classes, so I think the methods are wrong here. This should be implemented using some kind of interface instead (even if it's default implemented in the API with those builders)

Reply:
These only exist to return the constants rather than creating a new builder each time.

Reply:
There should be a way to provide these optimizations without mixin API classes, e.g. Lantern might also want to do this.

Reply:
Feel free to suggest.

Reply:
@kashike As I said to use the builders, the builders themselves are fast enough if they default to returning the constant versions. No need for mixing.